### PR TITLE
EVG-2249 Add project variant/task aliases

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: model/project_aliases.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+var (
+	ProjectIDKey = bsonutil.MustHaveTag(ProjectAlias{}, "ProjectID")
+	AliasKey     = bsonutil.MustHaveTag(ProjectAlias{}, "Alias")
+	VariantsKey  = bsonutil.MustHaveTag(ProjectAlias{}, "Variants")
+	TasksKey     = bsonutil.MustHaveTag(ProjectAlias{}, "Tasks")
```
Comment (tychoish) [inline model/project_aliases.go]: is it plausible to make these private?

File: model/project_aliases.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+//
+// ALIAS                  VARIANTS          TASKS
+// github_pull_request    .*linux.*         .*test.*
+// github_pull_request    ^ubuntu1604.*$    ^comppile.*$
+//
+// This will cause a GitHub pull request to create and finalize a patch which runs
+// all tasks containing the string “test” on all variants containing the string
+// “linux”; and to run all tasks beginning with the string “compile” to run on all
+// variants beginning with the string “ubuntu1604”.
+type ProjectAlias struct {
```
Comment (tychoish) [inline model/project_aliases.go]: include the _id?

index changes?

File: model/project_aliases.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+func (p ProjectAlias) Insert() error {
+	err := db.Insert(ProjectAliasCollection, p)
+	if err != nil {
+		return errors.Wrap(err, "failed to insert project alias")
+	}
+	return nil
+}
+
+// Remove removes a matching project alias from the database.
+func (p ProjectAlias) Remove() error {
```
Comment (tychoish) [inline model/project_aliases.go]: pointer receiver

**Comment by tychoish** [comment]
... but I don't have a better idea.

File: model/project_aliases.go (hunk 1 above)
Comment (bsamek) [reply model/project_aliases.go]: done

File: model/project_aliases.go (hunk 2 above)
Comment (bsamek) [reply model/project_aliases.go]: done

File: model/project_aliases.go (hunk 3 above)
Comment (bsamek) [reply model/project_aliases.go]: It's a small struct, and we don't mutate it.

Comment (tychoish) [reply model/project_aliases.go]: strings may not e *that* small, presumably we are going to mutate it when people change/modify them (as they will want to do). 

I dunno

File: model/project_aliases.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	q := db.Query(bson.M{
+		projectIDKey: p.ProjectID,
+		aliasKey:     p.Alias,
+		variantsKey:  p.Variants,
+		tasksKey:     p.Tasks,
+	})
```
Comment (tychoish) [inline model/project_aliases.go]: I think give that the struct serializes, you can just pass it without copying its values to a map

also if you put the _id in this, you can just do a remove by id?

File: model/project_aliases.go (hunk 3 above)
Comment (bsamek) [reply model/project_aliases.go]: done

File: model/project_aliases.go (hunk 4 above)
Comment (bsamek) [reply model/project_aliases.go]: done

**Pull request closed without merging**